### ISSUE TITLE
Changing base colors

### ISSUE THREAD

**Issue by LightCannon** [zhuzichu520/FluentUI#128]
Hello. 
I think there should be a way to change the colors for different widgets. Currently, all takes colors from FluTheme which does not have a way to change its colors (as library/plugin user).

**Comment by LightCannon** [comment]
Yes, but what I mean is not this.This means that in order to change the
color palette you have to change every color in the widget on your own,
basically building a "theming" logic on your own

What I'm asking about is changing the base colors which are defined c++
side.

**Comment by parker-int64** [comment]
You mean the color defined in c++ side?

Changing them will have to rebuild the project to make it work.


https://github.com/zhuzichu520/FluentUI/blob/0dbbba56da52fb1752b5a2f16d23c4d0ceb0377d/src/FluColors.cpp#L13-L120

**Comment by suterberg** [comment]
> 

重写一下源码么,很简单的,用一个槽函数去读取json数据并赋值,给你个python的示例吧,现在正在用python重写该库
`@QmlNamedElement("FluColors")
@QmlSingleton
class FluColors(QObject):

    def __init__(self, par=None):
        super().__init__(parent=par)
        print(__file__)
        with open(Path(__file__).parent / "color.json", 'r') as f:
            self.m_schemes = json.load(f)
        self.m_activeScheme = {}
        self.m_activeSchemeName = "yellow"
        self.setScheme("yellow")
        self.sets = FluColorSet(self)

    @Signal
    def flucolorChanged(self):
        ...

    @Slot(str)
    def setScheme(self, theme):
        for k, v in self.m_schemes[theme].items():
            self.m_activeScheme[k] = QColor.fromString(v)
        self.m_activeSchemeName = theme
        self.flucolorChanged.emit()

    @Property(QColor, constant=True)
    def black(self):
        return QColor.fromString("#000000")

    @Property(QColor, constant=True)
    def white(self):
        return QColor.fromString("#ffffff")

    @Property(QColor, constant=True)
    def grey10(self):
        return QColor.fromString("#faf9f8")

    @Property(QColor, constant=True)
    def grey20(self):
        return QColor.fromString("#f3f2f1")

    @Property(QColor, constant=True)
    def grey30(self):
        return QColor.fromString("#edebe9")

    @Property(QColor, constant=True)
    def grey40(self):
        return QColor.fromString("#e1dfdd")

    @Property(QColor, constant=True)
    def grey50(self):
        return QColor.fromString("#d2d0ce")

    @Property(QColor, constant=True)
    def grey60(self):
        return QColor.fromString("#c8c6c4")

    @Property(QColor, constant=True)
    def grey70(self):
        return QColor.fromString("#beb9b8")

    @Property(QColor, constant=True)
    def grey80(self):
        return QColor.fromString("#b3b0ad")

    @Property(QColor, constant=True)
    def grey90(self):
        return QColor.fromString("#a19f9d")

    @Property(QColor, constant=True)
    def grey100(self):
        return QColor.fromString("#979592")

    @Property(QColor, constant=True)
    def grey110(self):
        return QColor.fromString("#8a8886")

    @Property(QColor, constant=True)
    def grey120(self):
        return QColor.fromString("#797775")

    @Property(QColor, constant=True)
    def grey130(self):
        return QColor.fromString("#605e5c")

    @Property(QColor, constant=True)
    def grey140(self):
        return QColor.fromString("#484644")

    @Property(QColor, constant=True)
    def grey150(self):
        return QColor.fromString("#3b3a39")

    @Property(QColor, constant=True)
    def grey160(self):
        return QColor.fromString("#323130")

    @Property(QColor, constant=True)
    def grey170(self):
        return QColor.fromString("#292827")

    @Property(QColor, constant=True)
    def grey180(self):
        return QColor.fromString("#252423")

    @Property(QColor, constant=True)
    def grey190(self):
        return QColor.fromString("#201f1e")

    @Property(QColor, constant=True)
    def grey200(self):
        return QColor.fromString("#1b1a19")

    @Property(QColor, constant=True)
    def grey210(self):
        return QColor.fromString("#161514")

    @Property(QColor, constant=True)
    def grey220(self):
        return QColor.fromString("#11100f")

    @Property(QColor, notify=flucolorChanged)
    def darkest(self):
        return self.m_activeScheme["darkest"]

    @Property(QColor, notify=flucolorChanged)
    def darker(self):
        return self.m_activeScheme["darker"]

    @Property(QColor, notify=flucolorChanged)
    def dark(self):
        return self.m_activeScheme["dark"]

    @Property(QColor, notify=flucolorChanged)
    def normal(self):
        return self.m_activeScheme["normal"]

    @Property(QColor, notify=flucolorChanged)
    def lightest(self):
        return self.m_activeScheme["lightest"]

    @Property(QColor, notify=flucolorChanged)
    def lighter(self):
        return self.m_activeScheme["lighter"]

    @Property(QColor, notify=flucolorChanged)
    def light(self):
        return self.m_activeScheme["light"]`
调用setsheame 就会改变darker,darkerst 等的颜色,c++的话参照修改一下就可以了
